starky/src/prover.rs

```diff
@@ -70,63 +69,113 @@ where
-            &mut challenger,
-            config.num_challenges,
-            stark.permutation_batch_size(),
-        );
-        let permutation_z_polys = compute_permutation_z_polys::<F, S, D>(
-            &stark,
-            config,
-            &trace_poly_values,
-            &permutation_challenge_sets,
-        );
+    let constraint_degree = stark.constraint_degree();
+    let lookups = stark.lookups();
+    let lookup_challenges = stark.uses_lookups().then(|| {
+        get_grand_product_challenge_set(&mut challenger, config.num_challenges)
+            .challenges
+            .iter()
+            .map(|ch| ch.beta)
+            .collect::<Vec<_>>()
+    });
+
+    let num_lookup_columns = lookups
+        .iter()
+        .map(|l| l.num_helper_columns(constraint_degree))
+        .sum();
 
-        let permutation_zs_commitment = timed!(
+    let auxiliary_polys_commitment = stark.uses_lookups().then(|| {
+        let lookup_helper_columns = timed!(timing, "compute lookup helper columns", {
+            let challenges = lookup_challenges.as_ref().expect("We do have challenges.");
+            let mut columns = Vec::with_capacity(num_lookup_columns);
+            for lookup in &lookups {
+                for &challenge in challenges {
+                    columns.extend(lookup_helper_columns(
+                        lookup,
+                        &trace_poly_values,
+                        challenge,
+                        constraint_degree,
+                    ));
+                }
+            }
+            columns
+        });
+
+        // Get the polynomial commitments for all auxiliary polynomials.
+        let auxiliary_polys_commitment = timed!(
             timing,
             "compute permutation Z commitments",
             PolynomialBatch::from_values(
-                permutation_z_polys,
+                lookup_helper_columns,
                 rate_bits,
                 false,
                 config.fri_config.cap_height,
                 timing,
                 None,
             )
         );
-        (permutation_zs_commitment, permutation_challenge_sets)
+
+        auxiliary_polys_commitment
     });
-    let permutation_zs_commitment = permutation_zs_commitment_challenges
-        .as_ref()
-        .map(|(comm, _)| comm);
-    let permutation_zs_cap = permutation_zs_commitment
+
+    let auxiliary_polys_cap = auxiliary_polys_commitment
         .as_ref()
         .map(|commit| commit.merkle_tree.cap.clone());
-    if let Some(cap) = &permutation_zs_cap {
+    if let Some(cap) = &auxiliary_polys_cap {
         challenger.observe_cap(cap);
     }
 
     let alphas = challenger.get_n_challenges(config.num_challenges);
-    let quotient_polys = compute_quotient_polys::<F, <F as Packable>::Packing, C, S, D>(
-        &stark,
-        &trace_commitment,
-        &permutation_zs_commitment_challenges,
-        public_inputs,
-        alphas,
-        degree_bits,
-        config,
+
+    #[cfg(test)]
+    {
```

Comment:
Not a fan of abusing `test` mode for this. If we can, at some point, we should investigate the actual impact of doing `log::debug!` for these kinds of cases. I get why it's here, and it's needed for sure, I just don't think `test` is the way to go here.

Reply:
I kind of disagree here. Verbosity level should not over impact runtime (apart from the overhead incurred by extra I/O). This `check_constraints` is used for testing and debugging, but has a serious impact on prover performances, and should never be run outside of testing purposes I think. (adds 45% proving time for erc20 on my machine).

Reply:
What I mean is this: we should check on what the overhead of e.g. `log::debug!("bla bla {}", expensive_calculation_here());` actually is when running with debug logging **off**. I am 99% sure that `expensive_calculation_here()` will not run. If am wrong, and the expensive computation does run, we should use something like the static debug logging filters that both `log` and `tracing` crates provide.

If that turns out to be the wrong path then it'd be better to have a proper feature, e.g. `expensive_debugging`, rather than using `test`.

Reply:
Other places are affected with similar logic, hence I think it's kind of out of scope of this PR (which just moves stuff around really). But a conditional feature flag may be better indeed.

Reply:
Before we add a feature I'd like to see hard proof that `log::debug!("bla bla {}", expensive_calculation_here());` actually is slow when LOG_LEVEL excludes debug logs. :)

Reply:
My concern was not about it being not run at a different log level. My concern was that the lowest level is TRACE (AFAIK) which we use for debugging failures. And I'd rather not clog debugging by adding something that crazily increases proving time, while trace logging already incurs a large overhead because of all the I/O operations (we're talking GBs file)